### PR TITLE
vllm/llama-vision llava calibration infinite loop fix 

### PR DESCRIPTION
**Pull request by WeiweiZhang1**

### CODE REVIEW

File: test/test_mllm.py [hunk 1]
```diff
@@ -80,8 +80,18 @@ def test_quant_block_names(self):
...
         
+    def test_quant_block_names(self):
+        from auto_round.mllm.mllm_dataset import MLLM_DATASET
+        class Myclass:
+            model_type=None
+        dataset = MLLM_DATASET['liuhaotian/llava'](template=Myclass(), model=None, tokenzier=None, dataset_path="liuhaotian/llava", seqlen=32, nsamples=32)
+        self.assertEqual(len(dataset.questions), 32)
+        dataset = MLLM_DATASET['liuhaotian/llava'](template=Myclass(), model=None, tokenzier=None, dataset_path="liuhaotian/llava", seqlen=2048, nsamples=512)
+        self.assertEqual(len(dataset.questions), 512)
+
```
Comment (wenhuach21) [inline test/test_mllm.py:91]: help set quant_nontext_module=True in test_quant_vision

File: auto_round/mllm/mllm_dataset.py [hunk 2]
```diff
@@ -122,18 +123,17 @@ def _check(questions, min_seqlen, max_seqlen, nsamples):
...
-                if seqlen > max_len:
-                    logger.warning(f"seqlen={seqlen} is greater than the max length of dataset {max_len},"
-                                   f" will change seqlen to {max_len - 128}")
-                    new_min_seqlen = max_len - 128
-                else:
-                    logger.warning(f"no enough sample for seqlen greater than {min_seqlen},"
-                                   f" will decrease to {min_seqlen - 128}")
-                    new_min_seqlen = min_seqlen - 128
-                return new_questions + _check(questions, new_min_seqlen, min_seqlen, nsamples - len(new_questions))
+                logger.warning(f"no enough sample for seqlen greater than {min_seqlen},"
```
Comment (wenhuach21) [inline auto_round/mllm/mllm_dataset.py]: If there are not enough samples, log a warning. If the insufficient samples result from pre-filtering based on word count, reduce the word count threshold and re-tokenize the new samples to verify if their sequence length meets the requirement. No need to log warning in this process. If there are no insufficient samples of seqlen, don't change the seqlen automatically

File: auto_round/mllm/autoround_mllm.py [hunk 3]
```diff
@@ -155,14 +159,15 @@ def __init__(
         if nsamples % batch_size != 0:
             nsamples = (nsamples // batch_size + 1) * batch_size
             logger.warning(f"'nsamples' is not divisible by 'batch_size', will adjusted to {nsamples}")
-
-        if quant_nontext_module or (dataset in CALIB_DATASETS.keys() and not _only_text_test(model, tokenizer)):
+            
+        if not isinstance(dataset, torch.utils.data.DataLoader) and (
+                quant_nontext_module or (dataset in CALIB_DATASETS.keys() and not _only_text_test(model, tokenizer))):
             if quant_nontext_module:
                 logger.warning(f"Quantitative nontext module is not supported for plain text datasets,"
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py]: text only dataset cannot be used for calibrating non-text modules, switching to liuhaotian/llava_conv_58k

File: auto_round/mllm/autoround_mllm.py [hunk 4]
```diff
@@ -155,14 +159,15 @@ def __init__(
...
-                               "will use liuhaotian/llava_conv_58k with default config as an alternative.")
+                            "will use liuhaotian/llava_conv_58k with default config as an alternative.")
             else:
                 logger.warning(f"{model.config.model_type} not support for {dataset},"
-                               " will use liuhaotian/llava_conv_58k with default config as an alternative.")
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py:165]: similar issue

**Comment by wenhuach21** [comment]
remember to remind user to include this fix in the recipe of llama-vision

File: test/test_mllm.py [hunk 5]
```diff
@@ -80,8 +80,19 @@ def test_quant_block_names(self):
         blocks = find_matching_blocks(model, all_blocks, to_quant_block_names)
         assert target_blocks == blocks
         
+    def test_dataset_check(self):
+        from auto_round.mllm.mllm_dataset import MLLM_DATASET
+        class Myclass:
+            model_type=None
+        dataset = MLLM_DATASET['liuhaotian/llava'](template=Myclass(), model=None, tokenzier=None, dataset_path="liuhaotian/llava", seqlen=32, nsamples=32)
+        self.assertEqual(len(dataset.questions), 32)
+        dataset = MLLM_DATASET['liuhaotian/llava'](template=Myclass(), model=None, tokenzier=None, dataset_path="liuhaotian/llava", seqlen=2048, nsamples=512)
```
Comment (wenhuach21) [inline test/test_mllm.py:89]: @n1ck-guo  you may need to have a check of this test on your pr

Comment (n1ck-guo) [reply test/test_mllm.py:89]: i will work on this pr, cause the new push of weiwei retriggered the ut.

File: auto_round/script/mllm.py [hunk 6]
```diff
@@ -229,6 +230,16 @@ def setup_lmeval_parser():
     return args
 
 
+def _default_args_check(args):
+    if 'liuhaotian/llava' in args.dataset:
+        args.truncation = False if "--trancation" not in sys.argv else args.truncation
+        args.batch_size = 1 if "--batch_size" not in sys.argv else args.batch_size
+        args.gradient_accumulate_steps = 4 if \
```
Comment (wenhuach21) [inline auto_round/script/mllm.py]: why use 4, default is 8

File: auto_round/script/mllm.py [hunk 7]
```diff
@@ -229,6 +230,16 @@ def setup_lmeval_parser():
     return args
 
 
+def _default_args_check(args):
+    if 'liuhaotian/llava' in args.dataset:
+        args.truncation = False if "--trancation" not in sys.argv else args.truncation
```
Comment (wenhuach21) [inline auto_round/script/mllm.py]: better move to API

File: auto_round/script/mllm.py [hunk 8]
```diff
@@ -229,6 +230,16 @@ def setup_lmeval_parser():
...
+        args.batch_size = 1 if "--batch_size" not in sys.argv else args.batch_size
+        args.gradient_accumulate_steps = 4 if \
+            "--gradient_accumulate_steps" not in sys.argv else args.gradient_accumulate_steps
+        args.seqlen = 512 if "--seqlen" not in sys.argv else args.seqlen
```
Comment (wenhuach21) [inline auto_round/script/mllm.py]: why need these two checker, the original code should already handle these issues

File: auto_round/script/mllm.py (hunk 7 above)
Comment (wenhuach21) [reply auto_round/script/mllm.py]: The entry should focus solely on tasks such as model loading, quantization layer configuration, and evaluation. All other functionalities should be shifted to the API to maintain consistency with both API and command-line usage.

File: auto_round/mllm/autoround_mllm.py [hunk 9]
```diff
@@ -171,7 +172,7 @@ def __init__(
             dataset = "liuhaotian/llava_conv_58k"
             self.truncation = False
             batch_size = 1
-            gradient_accumulate_steps = 4
+            gradient_accumulate_steps = 8
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py]: batch-size, seqlen, gradient_accumualte_steps all follow user's setting

File: auto_round/mllm/autoround_mllm.py [hunk 10]
```diff
@@ -170,10 +166,15 @@ def __init__(
...
                             " will use liuhaotian/llava_conv_58k with default config as an alternative.")
             dataset = "liuhaotian/llava_conv_58k"
-            self.truncation = False
+
+        if dataset not in CALIB_DATASETS.keys():
+            truncation = False
             batch_size = 1
-            gradient_accumulate_steps = 8
-            seqlen = 512
+            seqlen = min(512, seqlen) if seqlen is not None else 512
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py]: if seqlen is None,  logger.waring and reset it to 512.  otherwise, just use the seqlen, no min op is needed

Comment (wenhuach21) [reply auto_round/mllm/autoround_mllm.py]: sorry, reset it to 2048

Comment (n1ck-guo) [reply auto_round/mllm/autoround_mllm.py]: seqlen is None means that user do not specify the seqlen. so default 512 for llava and 2048 for text-only.

File: auto_round/mllm/autoround_mllm.py [hunk 11]
```diff
@@ -172,7 +172,10 @@ def __init__(
                 truncation = False
                 batch_size = 1
                 seqlen = 512 if seqlen is None else seqlen
-
+        if quant_nontext_module and batch_size != 1:
+            logger.warning(f"batch_size({batch_size}) cannot be used for calibrating non-text modules,"
+                           "reset to 1")
+            batch_size = 1
```
Comment (wenhuach21) [inline auto_round/mllm/autoround_mllm.py:179]: the gradient_accumualte should be changed too

**Review by wenhuach21 — APPROVED** [review]

**Pull request merged**